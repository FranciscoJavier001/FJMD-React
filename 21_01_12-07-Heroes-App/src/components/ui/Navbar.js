//**_______________________________________________________________________________________________________________________________________________*/
//** Esto es un Navbar - Lo copie del Git */
import React from 'react'
import { Link, NavLink } from 'react-router-dom'


//  export const Navbar = ({ history }) => { //** Aqui en pego el history, no compartir mucha informacion por los props */
export const Navbar = () => { //** Es la linea 8, pero remplazo la sintaxis */


    return (
        //** navbar=Para que todo este junto y no haya espacios */
        //** navbar-expand-sm=para poner el navbar en fila y separados */
        //** navbar-dark=letras en color gris */
        //** bg-dark=fondo en negro */
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" //** Es para que salga al inicio y mas grande */
                to="/" //** Donde me lleva cuando hago click */
            >
                HeroesApp {/* Texto */}
            </Link>

            <div className="navbar-collapse col-1"> {/* Hacer mas chico la anchura */}
                <div className="navbar-nav">{/* Para que todas esten en hileras */}

                    <NavLink 
                        activeClassName="active" //** Dond este seleccionado se quede en color blanco */
                        className="nav-item nav-link" //** nav-item=Item de navegacion - nav-link= es para que todos se vean a la altura */
                        exact //** Es la direccion a la que lleva */
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink //** Ruta que me lleva al Search en el NavBar */
                        activeClassName="active" //** Para que se quede blanco cuando este ahi */
                        className="nav-item nav-link"
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

          
        </nav>
    )
}