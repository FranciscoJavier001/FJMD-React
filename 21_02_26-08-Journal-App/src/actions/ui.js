//** Acciones en el UI */

import { types } from "../types/types";

export const setError = ( err ) => ({ //** De una vez voy a hacer el return del objeto y va a recibir el payload del error */
    type: types.uiSetError,
    payload: err
})

export const removeError = () => ({ //** De una vez voy a hacer el return del objeto y no va a recibir ningun objeto */
    type: types.uiRemoveError, //** Va a llamar a la funcion de remover error */
})

export const startLoading = () => ({ //** Asi llamamos las acciones, y ahora se debe hacer el dispatch de estas acciones */
    type: types.uiStartLoading,
})

export const finishLoading = () => ({
    type: types.uiFinishLoading,
})