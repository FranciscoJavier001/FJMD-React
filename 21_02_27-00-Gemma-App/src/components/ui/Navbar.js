import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types';

// export const Navbar = ({ history }) => { //** Aqui pegamos el history, como en la pagina de LoginScreen, pero es importante no compartir mucha informaciona traves de las propertyes y que los componentes sean mas sencillos y limpios */

export const Navbar = () => { //** Esta es la que estaba en la linea 6, pero ahora la vamos a remplazar de otra manera */

    //** 1. Para esta tarea voy a necesitar acceso al context, en el contextValue voy a ponerle el AuthContext y en el context voy a extraer el user, y en el user voy a extraer el name del objeto */
    //** Para la tarea del logout hay que hacer el dispatch y purgar la informacion que tenemos en el reducer */
    const { user:{ name }, dispatch } = useContext( AuthContext );
    //** react-router-dom nos ofrece un useHistory */
    const history = useHistory(); //** Este me regresa un history y asi de sencillo soluciono el problema de compartir la info a traves de las props */

    // console.log(history);
    // console.log(name);

    const handleLogout = () => { //** Esta es la funcion que le vamos a dar al boton */
        //** Llamamos el dispatch, mandamos una accion que tiene el type y ese types viene del .logout, esto es la primera parte de la tarea, asi se borra todo y el logged en false */
        //** Con esto hacemos que primero navege, despues purge, atencion el as seguientes lineas */
        //** Llamamos el history.replace, para que ya no mantenga esta pagina (es decir para que cambie la navegcion y que se lanze al login), entonces el NavBar ya esta recibiendo el history y por consecuencia aqui ya puedo poner el replace para que funcione */

        // console.log('Click!'); //** Este se va a hacer cuando se de un click, osea en la prueba si se simula el click de esa funcion */

        history.replace('/login');

        dispatch({
            type: types.logout
        });
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                KANDICHI
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">
{/* 
Masaje Reductivo 10 sesiones + 1 pedicura de regalo 2999
Pedicura Febrero 2x1 380
Febrero 550
Terapia Reiki - Es una tecnica de canalizacion y transmision de energia vital a traves de la imposicion de manos, que se utiliza para obtener paz y equilibrio en todos los niveles: fisico, mental, emocional y espiritual 300
Pedicura Spa 350
Callosidades 280
Uñas Enterradas 300
Reconstruccion de uñas 400
Hidratacion Extrema Agotado
Resequedad Extrema 290
Preventivo 200
Ventosas calientes Bebida y parches de colageno de cortesia 300
Masaje Incluimos bebida de cortesia y parches de colageno de hidratacion extrema
Hongos bye
*/}
                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        Servicios
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        Masajes
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>

                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ml-auto">
                    
                    {/* Aqui voy a crear un parrafo que me diga un nombre */}
                    {/* Otra tarea, que extraigamos el context con el nombre y lo mostremos aqui, esto lo vamos a tener que reemplazar */}
                    <span className="nav-item nav-link text-info">
                        { name }
                    </span>

                    {/* Que cuando toquemos el boton nos saque con el { handleLogout }, pero lo declaramos, que sea una funcion de flecha que no recibe ningun parametro y lo que hace es el dispatch de la accion respectiva que es el logout usando los types y asegurarnos que en el estado de la aplicacion no tengamos el nombre, es decir que se haya borrado y que tenga el logged en false, echo eso significa que paso por el reducer, que quitamos la identificacion del usuario, que se guardo en el localStorage porque se disparo el efecto, en cuestion es solo disparar la accion y que navegemos a la pagina del login */}
                    <button 
                        className="nav-item nav-link btn"
                        onClick={ handleLogout }
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}