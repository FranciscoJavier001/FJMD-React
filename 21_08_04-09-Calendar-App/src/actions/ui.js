//** Estos son los endpoints */
import { types } from "../types/types";

//** Este esta en types>types */
export const uiOpenModal = () => ({ type: types.uiOpenModal }); //** Solo va a regresar un objeto */
export const uiCloseModal = () => ({ type: types.uiCloseModal }); //** Solo va a regresar un objeto */