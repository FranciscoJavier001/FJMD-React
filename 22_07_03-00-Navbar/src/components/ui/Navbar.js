import { BrowserRouter, NavLink } from "react-router-dom";
import React from 'react'

export const Navbar = () => {
  return (
    <BrowserRouter>

      <nav className="navbar navbar-expand-lg navbar-dark bg-dark">

        <NavLink
          className="navbar-brand"
          to="/"
        >
          Principal
        </NavLink>

        <div className="navbar-collapse col-1">
          <div className="navbar-nav">

            <NavLink
              to="/"
              className={({isActive}) => isActive ? 'nav-item active' : 'nav-item'}
            >Primero
            </NavLink>
          </div>
        </div>

      </nav>
    </BrowserRouter>
  )
}
