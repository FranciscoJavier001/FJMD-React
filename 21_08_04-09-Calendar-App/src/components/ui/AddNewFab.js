//** ________________________________________________________________________________________________________________________________________________ */
import React from 'react'
import { useDispatch } from 'react-redux'
import { uiOpenModal } from '../../actions/ui';

export const AddNewFab = () => {

    const dispatch = useDispatch();

    const handleClickNew = () => { //** No recibo ningun argumento */
        dispatch( uiOpenModal() ) //** Y el Dispatch solo va a ser el uiOpenModal reducers>uiReducer */
    }

    return (
        <button
            className="btn btn-primary fab"
            onClick={ handleClickNew }//** Asocio el metodo en el onClick */
        >
            <i className="fas fa-plus"></i>
        </button>
    )
}
