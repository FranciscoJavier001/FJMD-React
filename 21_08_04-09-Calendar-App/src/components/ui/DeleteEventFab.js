//**_______________________________________________________________________________________________________________________________________________*/
import React from 'react'
import { useDispatch } from 'react-redux'
import { eventStartDelete } from '../../actions/events';

export const DeleteEventFab = () => {

    const dispatch = useDispatch();

    const handleDelete = () => {
        // console.log('click'); //** Hacer esto para confirmar que se llame haciendo click en los test */
        dispatch( eventStartDelete() ) //** Esta la hicimos en los events y borramos la pasada, que la dejamos local */
    }

    return (
        <button
            className="btn btn-danger fab-danger"
            onClick={ handleDelete }
        >
            <i className="fas fa-trash"></i>
            <span> Borrar Evento </span>
        </button>
    )
}
