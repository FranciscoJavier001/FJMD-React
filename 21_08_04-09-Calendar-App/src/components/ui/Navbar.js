//**_______________________________________________________________________________________________________________________________________________*/
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth'

export const Navbar = () => {

    const dispatch = useDispatch()
    const { name } = useSelector( state => state.auth ) //** Desestructuramos el name del state */

    const handleLogout = () => { //** Funcion que no recibe argumentos y hace el dispatch del logout */
        dispatch( startLogout() ) //** Funcion esta en actions>auth y hace las acciones ahi definidas */
        // console.log('click'); //** Asi confirmo que este funcionando el boton */
    }

    return (
        <div className="navbar navbar-dark bg-dark mb-4"> {/* Esto es el navbar, con estilos */}
            <span className="navbar-brand">
                { name } {/* Con useSelector, va arriba del state sacamos el nombre y lo mostramos */}
            </span>

            <button
                className="btn btn-outline-danger" /* btn(bootstrap), outline dibujado por fuera, danger color rojo */
                onClick={ handleLogout } //** Al hacer click llamo a este metodo */
                >
            <i className="fas fa-sign-out-alt"></i> {/* i=icon, Puerta de salida FA */}
                <span> Salir</span>
            </button>
        </div>
    )
}
