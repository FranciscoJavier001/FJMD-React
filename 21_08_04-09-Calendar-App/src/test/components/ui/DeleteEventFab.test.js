//**_______________________________________________________________________________________________________________________________________________*/
import Enzyme from 'enzyme'; //** Si no tengo estos me manda falla */
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; //** Si no tengo estos me manda falla */

Enzyme.configure({ adapter: new Adapter() }); //** Si no tengo estos me manda falla */

import React from 'react' //** Uso React porque voy a renderizar un componente */
import { mount } from 'enzyme'
import { Provider } from 'react-redux' //** Para utilizar el mount necesito un componente llamado Provider */

import configureStore from 'redux-mock-store' //** Para configurar el store */
import thunk from 'redux-thunk' //** La de arriba ocupa esta porque las acciones retornan una funcion */

import '@testing-library/jest-dom' //** Ayuda con el tipado */
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab'

const middlewares = [ thunk ] //** Funcion que se invoca despues de que se envia una accion, puede modificarla, esperar que termine o cancelarla */
const mockStore = configureStore( middlewares ) //** MockStore es un objero que simula ser otro y el store configura funciones de los moddleware */

const initState = {} //** Configuro el iS, que asi quiero cuando inicie las pruebas */
const store = mockStore( initState ) //** Este va con el berofeEach para inicializar todas las acciones que este store haya ejecutado */
store.dispatch = jest.fn() //** Puedo saber con que argumento se utilizo para llamar el dispatch, por eso lo manejo con un mock */

const wrapper = mount( //** Dentro necesito el componente llamado Provider */
    <Provider store={ store }> {/* Esto me va a proveer el store, pero debo de crearlo, dentro mando el store creado anteriormente */}
        <DeleteEventFab />{/* Dentro de este HOC voy a colocar el evento a analizar */}
    </Provider>
)

describe('Pruebas en <DeleteEventFab />', () => {
    
    test('debe de mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot() //** Para que haga match con el snapshot */
    })
})