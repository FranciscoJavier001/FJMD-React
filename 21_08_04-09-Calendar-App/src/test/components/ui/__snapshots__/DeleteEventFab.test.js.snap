// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DeleteEventFab /> debe de mostrarse correctamente 1`] = `ReactWrapper {}`;
